pkg/operator/reports_test.go

```diff
@@ -92,3 +102,235 @@ func TestGetNextReportPeriod(t *testing.T) {
+	reportGetter := testhelpers.NewReportStore(nil)
+
+	reportStart := &time.Time{}
+	reportEndTmp := reportStart.AddDate(0, 1, 0)
+	reportEnd := &reportEndTmp
+
+	testTable := []struct {
+		name         string
+		report       *metering.Report
+		expectErr    bool
+		expectErrMsg string
+	}{
+		{
+			name:         "empty spec.generationQuery returns err",
+			report:       testhelpers.NewReport(testReportName, testNamespace, "", reportStart, reportEnd, v1alpha1.ReportStatus{}, nil, false),
+			expectErr:    true,
+			expectErrMsg: "must set spec.generationQuery",
+		},
+		{
+			name:         "spec.ReportingStart > spec.ReportingEnd returns err",
+			report:       testhelpers.NewReport(testReportName, testNamespace, testNonExistentQueryName, reportEnd, reportStart, v1alpha1.ReportStatus{}, nil, false),
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("spec.reportingEnd (%s) must be after spec.reportingStart (%s)", reportStart.String(), reportEnd.String()),
+		},
+		{
+			name:         "spec.ReportingEnd is unset and spec.RunImmediately is set returns err",
+			report:       testhelpers.NewReport(testReportName, testNamespace, testNonExistentQueryName, reportStart, nil, v1alpha1.ReportStatus{}, nil, true),
+			expectErr:    true,
+			expectErrMsg: "spec.reportingEnd must be set if report.spec.runImmediately is true",
+		},
+		{
+			name:         "spec.GenerationQueryName does not exist returns err",
+			report:       testhelpers.NewReport(testReportName, testNamespace, testNonExistentQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{}, nil, false),
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("ReportGenerationQuery %s does not exist", testNonExistentQueryName),
+		},
+		{
+			name:         "valid report with invalid DataSource returns error",
+			report:       testhelpers.NewReport(testReportName, testNamespace, testInvalidQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{}, nil, true),
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("failed to validate ReportGenerationQuery dependencies %s: %s", testInvalidQueryName, "ReportDataSource.metering.openshift.io \"this-does-not-exist\" not found"),
+		},
+		{
+			name:         "valid report with valid DataSource returns nil",
+			report:       testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{}, nil, true),
+			expectErr:    false,
+			expectErrMsg: "",
+		},
+	}
+
+	for _, testCase := range testTable {
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			_, _, err := validateReport(testCase.report, queryGetter, dataSourceGetter, reportGetter, &reporting.UninitialiedDependendenciesHandler{})
+
+			if testCase.expectErr {
+				assert.EqualErrorf(t, err, testCase.expectErrMsg, "expected that validateReport would return the correct error message")
+			} else {
+				assert.NoErrorf(t, err, "expected the report '%s' would return no error, but got '%v'", testCase.name, err)
```

Comment:
You shouldn't need to print the test case name since it's in the output already. Also, you refer to the test case name as the report name. Probably not necessary either since each test case uses the same `testReportName` as reportName.

Reply:
Updated.